packages/suite/src/support/suite/useConnectPopupWeb.tsx

```diff
@@ -31,12 +31,16 @@ export const useConnectPopupWeb = () => {
     const lifecycle = useSelector(state => state.suite.lifecycle);
     const manifest = useRef<ManifestPartial | undefined>(undefined);
     const [pendingHandshake, setPendingHandshake] = useState<string | undefined>();
+    const [responseSent, setResponseSent] = useState(false);
 
     useEffect(() => {
         const onMessage = async (event: MessageEvent) => {
             if (event.data?.type === 'channel-handshake-request') {
                 postMessageToParent({ type: 'channel-handshake-confirm' });
-            } else if (event.data.type === POPUP.HANDSHAKE) {
+            } else if (
+                event.data.type === POPUP.HANDSHAKE &&
+                event.data.payload?.settings?.manifest
```

Comment:
Just to clarify: This means that the manifest is required? Handshake event without manifest is not considered a proper handshake?